app/modules/entities/components/layouts/works_browser_facets.svelte

```diff
@@ -1,31 +1,39 @@
 <script>
   import SelectDropdown from '#components/select_dropdown.svelte'
-  import { i18n } from '#user/lib/i18n'
-  import { entityProperties, getWorksFacets } from '#entities/components/lib/works_browser_helpers'
-  import Spinner from '#components/spinner.svelte'
+  import { entityProperties, getWorksFacets, getFacetsSelectors } from '#entities/components/lib/works_browser_helpers'
   import { getContext } from 'svelte'
+  import { i18n, I18n } from '#user/lib/i18n'
 
   export let works, facets, facetsSelectors, facetsSelectedValues, flash
 
   const layoutContext = getContext('layout-context')
+  const facetsObj = getWorksFacets({ works, context: layoutContext })
 
-  const waitForFacets = getWorksFacets({ works, context: layoutContext })
-    .then(res => ({ facets, facetsSelectedValues, facetsSelectors } = res))
-    .catch(err => flash = err)
-</script>
+  let loadingMessage
+  const displayLoadingInfoMessage = async facetsObj => {
+    const urisSize = facetsObj.valuesUris.length
+    loadingMessage = I18n('loading_may_take_a_while', { smart_count: urisSize })
+  }
 
-{#await waitForFacets}
-  <Spinner />
-{:then}
-  {#each Object.keys(facets) as property}
-    {#if !facetsSelectors[property].disabled}
-      <SelectDropdown
-        bind:value={facetsSelectedValues[property]}
-        options={facetsSelectors[property].options}
-        resetValue="all"
-        buttonLabel={i18n(property)}
-        withImage={entityProperties.includes(property)}
-      />
-    {/if}
-  {/each}
-{/await}
+  ({ facets, facetsSelectedValues } = facetsObj)
```

Comment:
non-blocking, but I would like to stick to this convention to have a `;` before object destructive re-assignments that aren't in a one-liner block, as it prevents issues when moving the line around, without having the cognitive load to need to check what is the line above to know if this could mistakenly be interpreted as a function call. 
```suggestion
  ;({ facets, facetsSelectedValues } = facetsObj)
```

Reply:
done in 98884209d